### PR TITLE
Cleanup DNS decompression + fix warning messages being shown

### PR DESCRIPTION
**Pull request by gpotter2**
- fixes https://github.com/secdev/scapy/issues/1846
- cleanup dns
- fixes some warnings
- make UTscapy display raw data of the packets that have made the dissector crash, to be able to debug it externally

### CODE REVIEW

**Comment by gpotter2** [comment]
Ready to be reviewed

**Comment by gpotter2** [comment]
@guedou @p-l- Please review

File: scapy/layers/dns.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 from scapy.pton_ntop import inet_ntop, inet_pton
 
 
-def dns_get_str(s, p, pkt=None, _internal=False):
+def dns_get_str(s, p, pkt=None, _fullpacket=False):
```
Comment (guedou) [inline scapy/layers/dns.py]: I think that it is a good opportunity to rename this one letter variables.

Comment (gpotter2) [reply scapy/layers/dns.py]: `s` stands for string (pretty well used in Scapy)
`p` stands for pointer

Should I keep `s` and get `p` -> `pointer` (a bit long, it’s used a lot afterwards)

Comment (guedou) [reply scapy/layers/dns.py]: If this change is not too much trouble, it will improve the readability of the algorithm,

**Comment by guedou** [comment]
flake8 is not happy.

**Comment by gpotter2** [comment]
Dont worry, make flake8 happy :smile:

Should be mergeable

**Comment by gpotter2** [comment]
The side effect of this PR is that it will now print out packets that make the dissector crash. 

Currently, some TLS packets make the tests crash from time to time, we’ll be able to debug that out.

**Pull request merged**